module/data/item/fields/spellcasting-field.mjs

```diff
@@ -27,4 +28,37 @@ export default class SpellcastingField extends SchemaField {
     Object.entries(fields).forEach(([k, v]) => !v ? delete fields[k] : null);
     super(fields, { label: "DND5E.Spellcasting", ...options });
   }
+
+  /* -------------------------------------------- */
+  /*  Data Preparation                            */
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare data for this field. Should be called during the `prepareFinalData` stage.
+   * @this {ItemDataModel}
+   * @param {object} rollData  Roll data used for formula replacements.
+   */
+  static prepareData(rollData) {
+    this.spellcasting.preparation.max = simplifyBonus(this.spellcasting.preparation.formula, rollData);
+
+    // Temp method for determining spellcasting type until this data is available directly using advancement
+    if ( CONFIG.DND5E.spellcastingTypes[this.spellcasting.progression] ) {
+      this.spellcasting.type = this.spellcasting.progression;
+    } else this.spellcasting.type = Object.entries(CONFIG.DND5E.spellcastingTypes).find(([type, data]) => {
+      return !!data.progression?.[this.spellcasting.progression];
```

Comment:
`find` doesn't require a strictly boolean return. I'm surprised the linter does not complain actually, I thought I added that rule.

```suggestion
    } else this.spellcasting.type = Object.entries(CONFIG.DND5E.spellcastingTypes).find(([, { progression }]) => {
      return progression?.[this.spellcasting.progression];
```